java/src/com/ibm/streamsx/rest/Element.java

```diff
@@ -37,4 +42,31 @@ void setConnection(StreamsConnection connection) {
     public final String toString() {
         return pretty.toJson(this);
     }
+    
+    private Gson refreshJson;
+    
+    /**
+     * Refresh this element.
+     * 
+     * Attributes of this object are updated from current state
+     * though the Streams REST API.
+     * 
+     * @throws IOException Error communicating with Streams.
+     */
+    public void refresh() throws IOException {
+        String response = connection().getResponseString(self);
```

Comment:
not all objects fill in the "self" string, if we don't have one, we could end up throwing an exception (or we should just throw a specific "can't refresh this object" exception?